packages/x-data-grid/src/components/GridLoadingOverlay.tsx

```diff
@@ -25,20 +25,20 @@ export interface GridLoadingOverlayProps extends GridOverlayProps {
 const LOADING_VARIANTS: Record<
   GridLoadingOverlayVariant,
   {
-    component: React.ComponentType;
+    component: (rootProps: DataGridProcessedProps) => React.ComponentType;
```

Comment:
Nit: To make the functioning more aligned with the naming.
```suggestion
    getComponent: (rootProps: DataGridProcessedProps) => React.ComponentType;
```